cmd/operator-sdk/bundle/validate.go

```diff
@@ -15,89 +15,156 @@
-				log.Fatalf("Error to unpacking image: %v", err)
-			}
+	c.addToFlagSet(cmd.Flags())
 
-			log.Info("Validating bundle image format and contents")
+	return cmd
+}
 
-			if err = val.ValidateBundleFormat(dir); err != nil {
-				log.Fatalf("Bundle format validation failed: %v", err)
-			}
-			manifestsDir := filepath.Join(dir, bundle.ManifestsDir)
-			if err = val.ValidateBundleContent(manifestsDir); err != nil {
-				log.Fatalf("Bundle content validation failed: %v", err)
+func (c bundleValidateCmd) validate() error {
+	if (c.imageTag == "") == (c.directory == "") {
+		return fmt.Errorf("either --tag or --directory must be set, but not both")
+	}
+	return nil
+}
+
+func (c *bundleValidateCmd) addToFlagSet(fs *pflag.FlagSet) {
+	fs.StringVarP(&c.directory, "directory", "d", "",
+		"The directory where bundle manifests and metadata are located or, "+
+			"if --content-only is set, the path to a bundle. Either --directory or --tag must be set but not both")
+	fs.StringVarP(&c.imageTag, "tag", "t", "",
+		"The path of a registry to pull from, image name and its tag that present the bundle image "+
+			"(e.g. quay.io/test/test-operator:latest). Either --directory or --tag must be set but not both")
+	fs.StringVarP(&c.imageBuilder, "image-builder", "b", "docker",
+		"Tool to extract container images. One of: [docker, podman]")
+	fs.BoolVarP(&c.contentOnly, "content-only", "c", false, "Only validate bundle contents")
+}
+
+func (c bundleValidateCmd) run() (err error) {
+	// Set up directory, either the one passed to --directory or a temp dir
+	// used to unpack the image.
+	dir := c.directory
+	if c.imageTag != "" {
+		dir, err = ioutil.TempDir("", "bundle-")
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err = os.RemoveAll(dir); err != nil {
+				log.Errorf("error removing temp bundle dir: %v", err)
 			}
+		}()
+	}
 
-			log.Info("All validation tests have completed successfully")
+	// Set up logger.
+	fields := log.Fields{"bundle-dir": dir}
+	if c.imageTag != "" {
+		fields = log.Fields{"container-tool": c.imageBuilder}
+	}
+	logger := log.WithFields(fields)
+	if viper.GetBool(flags.VerboseOpt) {
+		log.SetLevel(log.DebugLevel)
+	}
 
-			return nil
-		},
+	val := bundle.NewImageValidator(c.imageBuilder, logger)
+
+	// Pull image if --tag was set.
+	if c.imageTag != "" {
+		logger.Info("Unpacked image layers")
+		err = val.PullBundleImage(c.imageTag, dir)
+		if err != nil {
+			logger.Fatalf("Error to unpacking image: %v", err)
+		}
 	}
 
-	cmd.Flags().StringVarP(&c.imageBuilder, "image-builder", "b", "docker",
-		"Tool to extract container images. One of: [docker, podman]")
+	// Validate bundle format if --content-only=false.
+	if c.contentOnly {
+		logger.Info("Validating bundle contents")
+	} else {
+		logger.Info("Validating bundle format and contents")
+		if err = val.ValidateBundleFormat(dir); err != nil {
+			logger.Fatalf("Bundle format validation failed: %v", err)
+		}
+	}
```

Comment:
Any reason we don't always want to validate the bundle format as well?

Reply:
With the changes in in #2715 this isn't necessary anymore. The original idea was to at least allow validation of a bundle directory even though `bundle create` was broken (see #2662).